packages/medusa/src/api-v2/admin/products/middlewares.ts

```diff
@@ -0,0 +1,146 @@
+  AdminPostProductsProductOptionsReq,
+  AdminPostProductsProductReq,
+  AdminPostProductsProductVariantsReq,
+  AdminPostProductsProductVariantsVariantReq,
+  AdminPostProductsReq,
+} from "./validators"
+import { transformBody, transformQuery } from "../../../api/middlewares"
+
+import { MiddlewareRoute } from "../../../loaders/helpers/routing/types"
+import { authenticate } from "../../../utils/authenticate-middleware"
+
+export const adminProductRoutesMiddlewares: MiddlewareRoute[] = [
+  {
+    method: ["ALL"],
+    matcher: "/admin/products*",
+    middlewares: [authenticate("admin", ["bearer", "session", "api-key"])],
+  },
+
+  {
+    method: ["GET"],
+    matcher: "/admin/products",
+    middlewares: [
+      transformQuery(
+        AdminGetProductsParams,
+        QueryConfig.listTransformQueryConfig
+      ),
+    ],
+  },
+  {
+    method: ["GET"],
+    matcher: "/admin/products/:id",
+    middlewares: [
+      transformQuery(
+        AdminGetProductsProductParams,
+        QueryConfig.retrieveTransformQueryConfig
+      ),
+    ],
+  },
+  {
+    method: ["POST"],
+    matcher: "/admin/products",
+    middlewares: [transformBody(AdminPostProductsReq)],
+  },
+  {
+    method: ["POST"],
+    matcher: "/admin/products/:id",
+    middlewares: [transformBody(AdminPostProductsProductReq)],
+  },
+  {
+    method: ["DELETE"],
+    matcher: "/admin/products/:id",
+    middlewares: [],
+  },
+
+  // TODO: Part of v1 API. Remove in favor of just updating the metadata through as standard update
+  {
+    method: ["POST"],
+    matcher: "/admin/products/:id/metadata",
+    middlewares: [],
+  },
+
+  {
+    method: ["GET"],
+    matcher: "/admin/products/:id/variants",
+    middlewares: [
+      transformQuery(
+        AdminGetProductsVariantsParams,
+        QueryConfig.retrieveTransformQueryConfig
+      ),
+    ],
+  },
+  // Note: New endpoint in v2
+  {
+    method: ["GET"],
+    matcher: "/admin/products/:id/variants/:variant_id",
+    middlewares: [
+      transformQuery(
+        AdminGetProductsProductVariantsVariantParams,
+        QueryConfig.retrieveTransformQueryConfig
+      ),
+    ],
+  },
+  {
+    method: ["POST"],
+    matcher: "/admin/products/:id/variants",
+    middlewares: [transformBody(AdminPostProductsProductVariantsReq)],
+  },
+  {
+    method: ["POST"],
+    matcher: "/admin/products/:id/variants/:variant_id",
+    middlewares: [transformBody(AdminPostProductsProductVariantsVariantReq)],
+  },
+  {
+    method: ["DELETE"],
+    matcher: "/admin/products/:id/variants/:variant_id",
+    middlewares: [],
+  },
+
+  // Note: New endpoint in v2
+  {
+    method: ["GET"],
+    matcher: "/admin/products/:id/options",
+    middlewares: [
+      transformQuery(
+        AdminGetProductsOptionsParams,
+        QueryConfig.retrieveTransformQueryConfig
+      ),
+    ],
+  },
+  // Note: New endpoint in v2
+  {
+    method: ["GET"],
+    matcher: "/admin/products/:id/options/:option_id",
```

Comment:
question: Do we need this?

Reply:
@olivermrbl I added it for consistency, as it adds very little overhead. I know the options model is pretty slim but I think it helps if we have consistency where possible